controllers/newscontroller.js

```diff
@@ -101,8 +100,9 @@ const paginateFeed = asyncHandler(async (req, res, next) => {
 
   const startIndex = (page - 1) * limit;
   const endIndex = page * limit;
-  const user = await User.findById(req.body.id);
+  const user = await News.findById(req.body.id);
```

Comment:
You're querying the entire news from News Entity. Pagination should be applied on the DB level. Find out how to filter News based on pagination

Reply:
ok sir i will try to paginate from db level